SignalServiceKit/src/Contacts/TSThread.m

```diff
@@ -395,16 +412,58 @@ - (BOOL)isMuted
             [mutedUntilDate timeIntervalSinceDate:now] > 0);
 }
 
-#pragma mark - Update With... Methods
+- (void)updateWithMutedUntilDate:(NSDate *)mutedUntilDate transaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    [self applyChangeToSelfAndLatestCopy:transaction
+                             changeBlock:^(TSThread *thread) {
+                                 [thread setMutedUntilDate:mutedUntilDate];
+                             }];
+}
+
+#pragma mark - Conversation Color
 
-- (void)updateWithMutedUntilDate:(NSDate *)mutedUntilDate
++ (NSString *)randomConversationColorName
 {
-    [self.dbReadWriteConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
-        [self applyChangeToSelfAndLatestCopy:transaction
-                                 changeBlock:^(TSThread *thread) {
-                                     [thread setMutedUntilDate:mutedUntilDate];
-                                 }];
-    }];
+    NSUInteger count = self.conversationColorNames.count;
+    NSUInteger index = arc4random_uniform((uint32_t)count);
+    return [self.conversationColorNames objectAtIndex:index];
+}
+
++ (NSString *)stableConversationColorNameForString:(NSString *)colorSeed
+{
+    NSData *contactData = [colorSeed dataUsingEncoding:NSUTF8StringEncoding];
+    
+    NSUInteger hashingLength = 8;
```

Comment:
`sizeof(unsigned long long)` instead of 8?

Reply:
done